theatre/studio/src/uiComponents/simplePointerMenu/useRequestPointerMenu.ts

```diff
@@ -0,0 +1,63 @@
+import type {$FixMe} from '@theatre/shared/utils/types'
+import {useCallback, useEffect, useState} from 'react'
+
+type IState = {isOpen: true; event: MouseEvent} | {isOpen: false}
+
+type CloseMenuFn = () => void
+
+export enum PointerMenuTrigger {
+  contextmenu,
+  mousedown,
+  click,
+  dblclick,
+}
+
+export const WHERE_LEFT_CLICK = (event: MouseEvent) => event.button === 0
+
+/** e.g. {@link WHERE_LEFT_CLICK} */
+export type PointerMenuFilter = (event: MouseEvent) => boolean
+
+export type IRequestPointerMenuOptions = {
+  trigger: PointerMenuTrigger
+  where?: PointerMenuFilter
+  disabled?: boolean
+}
+
+const useRequestPointerMenu = (
```

Comment:
Probably too early for this type of commnent, but want to make sure you know `usePopover()` exists.

Reply:
Yep, that was a much better approach. Sorry I missed it the first time.